### PR TITLE
[WIP] Docker-compose better setup and Jenkins pipeline

### PR DESCRIPTION
**Pull request by sandrew**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,6 @@ rvm:
 script:
   - 'bundle exec rails db:migrate RAILS_ENV=test'
   - 'bundle exec rspec spec'
-  - 'bundle exec rubocop'
```
Comment (ikapelyukhin) [inline .travis.yml]: Hmm, why?

Comment (sandrew) [inline .travis.yml]: because we have CodeClimate now. I turned it to be required for merge in settings

Comment (ikapelyukhin) [inline .travis.yml]: Rubocop is disabled in CodeClimate config. We're using our Rubocop rules gem, and CodeClimate can't handle that.

Comment (ikapelyukhin) [inline .travis.yml]: @sandrew Make Rubocop great again! :grimacing:

Comment (sandrew) [inline .travis.yml]: Ok

Comment (ikapelyukhin) [inline .travis.yml]: :grimacing::+1:

File: bin/rspec [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+#
+# This file was generated by Bundler.
```
Comment (ikapelyukhin) [inline bin/rspec]: And what is this? :-)

Comment (sandrew) [inline bin/rspec]: binstub for rspec

Comment (ikapelyukhin) [inline bin/rspec]: Hmm, but why?

File: README.md [hunk 3]
```diff
@@ -1,26 +1,16 @@
-# README
+# Repository Mirroring Tool
 ![Kartoha](https://travis-ci.org/SUSE/rmt.svg?branch=master)
 
+This tool allows you to mirror RPM update repositories in your own private network. Registered organization credentials are required to mirror SUSE repositories.
```
Comment (ikapelyukhin) [inline README.md]: Wording: `update repositories`. Please drop the word `update`.
There are Pool and Update repositories, and RMT can mirror them both! :grimacing:

Comment (ikapelyukhin) [inline README.md]: Better wording would be: credentials are required for SUSE Enterprise Linux repositories.
openSUSE repos are SUSE repos too.

File: README.md [hunk 4]
```diff
@@ -1,26 +1,16 @@
...
-* Database initialization
-
-* How to run the test suite
-
-* Services (job queues, cache servers, search engines, etc.)
-
-* Deployment instructions
-
-* ...
+The application is tested only on ruby 2.4.1 and newer. Support of older MRI is not intended.
```
Comment (ikapelyukhin) [inline README.md]: Hmm, please add `.ruby-version` while you are at it. :grimacing:

File: config/database.yml [hunk 5]
```diff
@@ -21,4 +21,4 @@ test:
 
 production:
   <<: *default
-  database: <%= ENV.fetch('POSTGRES_DB') { 'rmt_prod' } %>
+  database: <%= ENV.fetch('POSTGRES_DB') { 'rmt' } %>
```
Comment (ikapelyukhin) [inline config/database.yml]: Both development and production environment DBs are now named `rmt`.
Please rename dev DB to `rmt_devel` then.

File: docker-compose.yml [hunk 6]
```diff
@@ -8,34 +8,33 @@ services:
...
       service: base
       file: docker-compose-base.yml
     build: .
     environment:
-      - RAILS_ENV=test
       - POSTGRES_HOST=db
-    volumes:
-      - ./:/srv/www/rmt
+      - RAILS_ENV=production
+      - SECRET_KEY_BASE=8ea53ad3bc6c03923e376c8bdd85059c1885524947a7efe53d5e9c9d4e39861106ffd6a2ece82b803072ed701e6c960bade91644979e679416c5f255007237ae
```
Comment (ikapelyukhin) [inline docker-compose.yml]: I'm not sure if it's a good idea to keep it stored in the repo.

File: nginx/default.conf [hunk 7]
```diff
@@ -1,9 +1,5 @@
 server {
     listen 80 default_server;
-    listen 443 ssl;
-
-    ssl_certificate /etc/nginx/ssl/nginx.crt;
-    ssl_certificate_key /etc/nginx/ssl/nginx.key;
```
Comment (ikapelyukhin) [inline nginx/default.conf]: This portion should be removed from the config as well:
```
set $ssl off;
if ($scheme = https) {
    set $ssl on;
}
```

File: docker-compose.yml (hunk 6 above)
Comment (sandrew) [inline docker-compose.yml]: that's a copy of development key from `secrets.yml` already committed here. The workflow for production servers is different anyway: either with env variable or with new encrypted key from Rails 5.1

Comment (ikapelyukhin) [inline docker-compose.yml]: OK! :+1:

**Pull request closed without merging**